relay-filter/src/config.rs

```diff
@@ -192,6 +196,102 @@ impl LegacyBrowsersFilterConfig {
     }
 }
 
+/// Configuration for a generic filter.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct GenericFilterConfig {
```

Comment:
The fact that we need so many `pub(crate)` declarations in this crate tells me that the code structure is suboptimal. Ideally this crate would have a slim public interface with private submodules and a few public re-exports. Out of scope for this PR though, because it applies to the entire crate.

See [internal guidelines](https://www.notion.so/sentry/Relay-Code-and-Architecture-4fac481cf27741aba54d39594f1cae58?pvs=4#0505679040be41b78bb13086dec4319f).

Reply:
Exactly, I wanted to propose this refactor.